### PR TITLE
Enable azure openai engines

### PR DESCRIPTION
**Pull request by ecatkins**
Love the library!

I had to make modifications to get it to work with the Azure instances of OpenAI models that I use.

Primarily, I made it possible to pass an `engine` parameter to the LLMClassifier as well/instead of the `model` parameter [which is what the Azure instance requires](https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/chatgpt?pivots=programming-language-chat-completions).

I tried to do it in a way that minimized the # of edits on the existing codebase - if this is something you'd be interested in including, happy to make any changes to fit with your design principles.

```
import openai
openai.api_type = "azure"
openai.api_version = "2023-07-01-preview"
openai.api_base = "https://azure-openai-xxxxx.openai.azure.com/"  # Your Azure OpenAI resource's endpoint value.
openai.api_key = "<AZURE-OPENAI_API-KEY>"
evaluator = Factuality(engine="<Azure-Deployment-Name>", model=None)
```

### CODE REVIEW

File: py/autoevals/llm.py [hunk 1]
```diff
@@ -78,18 +78,21 @@ def __init__(
...
-        if not found:
-            raise ValueError(f"Unsupported model: {model}. Currently only supports OpenAI chat models.")
+        if engine is None:
+            for m in SUPPORTED_MODELS:
+                # Prefixes are ok, because they are just time snapshots
+                if model.startswith(m):
+                    found = True
+                    break
+            if not found:
+                raise ValueError(f"Unsupported model: {model}. Currently only supports OpenAI chat models.")
```
Comment (ankrgyl) [inline py/autoevals/llm.py:91]: We may want to just remove this check altogether — we definitely support more than the listed models, and "non-chat" flavored models are all deprecated at this point.

**Review by ankrgyl — APPROVED** [review]
This looks great! Just left one comment. Really appreciate the contribution 🙏

**Comment by ankrgyl** [comment]
One more thing -- can you make the equivalent change in the JS implementation?

File: py/autoevals/llm.py (hunk 1 above)
Comment (ecatkins) [reply py/autoevals/llm.py]: Removed

**Comment by ecatkins** [comment]
> One more thing -- can you make the equivalent change in the JS implementation?

JS isn't really my forte - I took a brief look - but chance I would do more harm than good :)

**Review by ankrgyl — APPROVED** [review]

**Comment by ankrgyl** [comment]
Hopefully last thing -- I ran the linter on your change and it appears to be failing (missing a comma after `engine`). I think you can fix this by running `make fixup`, or just add the comma in directly.

**Comment by ankrgyl** [comment]
Ah yep -- I think the way we cache parameters will be invalidated everytime a new one is added. Let me update the cached test results and push to your branch

**Pull request merged**